src/osm2pgsql.cpp

```diff
@@ -81,6 +81,10 @@ static void check_db(options_t const &options)
 
     init_database_capabilities(db_connection);
 
+    if (get_postgis_version().major < 3) {
+        throw std::runtime_error{"Need at least PostGIS version 3"};
+    }
```

Comment:
We don't need to check for PostGIS >= 3 as the SQL statement we've got will work on older versions. The order the old versions produce is not as good as GeoHash but it still works.

Reply:
Shouldn't that be `pv.minor < 5`?